packages/skeleton-react/src/lib/components/Rating/Rating.tsx

```diff
@@ -0,0 +1,196 @@
+        const fractionWidth = ratingRect.width / fraction;
+        const left = event.clientX - ratingRect.left;
+        let selectedFraction = Math.floor(left / fractionWidth) + 1;
+
+        if(getComputedStyle(figureRef.current).direction === 'rtl') {
+            selectedFraction = fraction - selectedFraction + 1;
+        }
+
+        value = order + selectedFraction / fraction;
+        onValueChange(value);
+        onMouseDown(event, value);
+    }, [fraction]);
+
+    const onRatingKeyDown = useCallback((event: React.KeyboardEvent<HTMLButtonElement>) => {
+        if(!figureRef.current) return;
+
+        const rtl = getComputedStyle(figureRef.current).direction === 'rtl';
+        switch(event.key) {
+            case 'ArrowLeft':
+                rtl ? increaseValue() : decreaseValue();
+                break;
+            case 'ArrowRight':
+                rtl ? decreaseValue() : increaseValue();
+                break;
+        }
+        onKeyDown(event);
+    }, []);
+
+    function refreshFocus() {
+        if(!figureRef.current) return;
+
+        const buttons = figureRef.current.querySelectorAll('button');
+        buttons[Math.max(0, Math.ceil(value - 1))].focus();
+    }
+
+    function increaseValue() {
+        value = Math.min(max, value + 1 / fraction);
+        onValueChange(value);
+        refreshFocus();
+    }
+
+    function decreaseValue() {
+        value = Math.max(0, value - 1 / fraction);
+        onValueChange(value);
+        refreshFocus();
+    }
+
+    return(
+        <>
+            <style>{`
+                .clip-left {
+                    clip-path: inset(0 0 0 var(--clip_value));
+                }
+                .clip-right {
+                    clip-path: inset(0 var(--clip_value) 0 0);
+                }
+            `}</style>
+            <figure ref={figureRef} className={`${base} ${width} ${text} ${fill} ${justify} ${spaceX} ${classes}`} data-testid="rating">
+                {[...Array(max)].map((_, order) => (
+                    <button
+                    className={`${buttonBase} ${buttonPosition} ${buttonAspect} ${buttonClasses} ${interactive ? undefined : 'pointer-events-none'}`}
+                    tabIndex={interactive && order === focusedButtonIndex ? 0 : -1}
+                    onMouseDown={(event: React.MouseEvent<HTMLButtonElement>) => interactive ? onRatingMouseDown(event, order) : undefined}
+                    onKeyDown={(event: React.KeyboardEvent<HTMLButtonElement>) => interactive ? onRatingKeyDown(event) : undefined}
+                    type='button'
+                    key={order}>
+                            <RatingContext.Provider value={{interactive, value, order, figureRef}}>
+                                {children}
+                            </RatingContext.Provider>
+                    </button>
+                ))}
+            </figure>
+        </>
+    );
+};
+
+const IconEmpty: React.FC<IconProps> = ({
```

Comment:
I think the dedicated `IconEmpty` and `IconFull` components may be overkill. For simple "slot-like" content (such as icons) React allows you to pass a `ReactNode` (read: React components themselves) as if they are props...

Types:
https://github.com/skeletonlabs/skeleton/blob/next/packages/skeleton-react/src/lib/components/Switch/types.ts#L63

Props:
https://github.com/skeletonlabs/skeleton/blob/next/packages/skeleton-react/src/lib/components/Switch/Switch.tsx#L42

Template Usage:
https://github.com/skeletonlabs/skeleton/blob/next/packages/skeleton-react/src/lib/components/Switch/Switch.tsx#L96

Example:
https://github.com/skeletonlabs/skeleton/blob/next/sites/next.skeleton.dev/src/examples/components/switch/ExampleIcons.tsx#L15

The end result will be a singular component that's much similar to the Svelte implementation! I realize this will take a non-trivial refactor, so let me know if you need help with this!